core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Run.java

```diff
@@ -48,241 +98,550 @@
+  public static final Long UNSAVED_ID = 0L;
+
+  public static Run createFromSolidXml(String statusXml, Function<String, SequencerReference> findSequencer) {
+    try {
+      Document statusDoc = SubmissionUtils.emptyDocument();
+      SubmissionUtils.transform(new UnicodeReader(statusXml), statusDoc);
+
+      Run run = new Run();
+      String runName;
+      if (statusDoc.getDocumentElement().getTagName().equals("error")) {
+        runName = (statusDoc.getElementsByTagName("RunName").item(0).getTextContent());
+      } else {
+        runName = (statusDoc.getElementsByTagName("name").item(0).getTextContent());
+        if (statusDoc.getElementsByTagName("name").getLength() != 0) {
+          for (int i = 0; i < statusDoc.getElementsByTagName("name").getLength(); i++) {
+            Element e = (Element) statusDoc.getElementsByTagName("name").item(i);
+            Matcher m = SOLID_RUN_REGEX.matcher(e.getTextContent());
+            if (m.matches()) {
+              runName = e.getTextContent();
+            }
+          }
+        }
+        String runStarted = statusDoc.getElementsByTagName("dateStarted").item(0).getTextContent();
+        DateFormat logDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+
+        run.setStartDate(logDateFormat.parse(runStarted));
+
+        if (statusDoc.getElementsByTagName("dateCompleted").getLength() != 0) {
+          String runCompleted = statusDoc.getElementsByTagName("dateCompleted").item(0).getTextContent();
+          run.setCompletionDate(logDateFormat.parse(runCompleted));
+        }
+      }
+
+      run.setAlias(runName);
+      run.setFilePath(runName);
+      run.setPairedEnd(false);
+
+      Matcher m = SOLID_RUN_REGEX.matcher(runName);
+      String instrument = null;
+      if (m.matches()) {
+        instrument = m.group(1);
+        run.setDescription(m.group(3));
+        if (m.group(3).startsWith("MP") || m.group(3).startsWith("PE")) {
+          run.setPairedEnd(true);
+        }
+      } else {
+        run.setDescription(runName);
+      }
+      if (statusDoc.getElementsByTagName("name").getLength() != 0) {
+        for (int i = 0; i < statusDoc.getElementsByTagName("name").getLength(); i++) {
+          Element e = (Element) statusDoc.getElementsByTagName("name").item(i);
+          Matcher me = SOLID_RUN_REGEX.matcher(e.getTextContent());
+          if (m.matches()) {
+            runName = e.getTextContent();
+            instrument = me.group(1);
+          }
+        }
+      }
+      run.setSequencerReference(findSequencer.apply(instrument));
+      run.setSecurityProfile(new SecurityProfile());
+      return run;
+    } catch (ParserConfigurationException | TransformerException | ParseException e) {
+      log.error("parse status XML", e);
+    }
+    return null;
+  }
+
+  private String accession;
+
+  @Column(nullable = false)
+  private String alias;
+
+  @OneToMany(targetEntity = RunChangeLog.class, mappedBy = "run", cascade = CascadeType.ALL)
```

Comment:
ChangeLogs are saved explicitly through ChangeLogStore, so this should not cascade persist. CascadeType.DELETE with orphan removal probably makes sense here, though we're not doing that anywhere else